### DISCUSSION TITLE
Change location of drawn electrons?

### DISCUSSION THREAD

**Discussion by bmcguir2** [rdkit/rdkit#4317] [Q&A]
Hello!  

I'm trying to implement a very simple use of the visualization capabilities of rdkit to display the structure of simple molecules.  Some of my molecules are radicals or have unpaired electrons, for example: CH+.

I can get the structure to display quite easily:

```Python3
mol = Chem.MolFromSmiles("[C+][H]")
mol
```
![image](https://user-images.githubusercontent.com/8889664/124937370-54488900-dfd5-11eb-9904-617a8440dd18.png)

The output (above) however has the electrons in a very misleading location; it implies they are associated with the lone hydrogen, rather than with the carbon atom, as they should be:

![image](https://user-images.githubusercontent.com/8889664/124937553-7c37ec80-dfd5-11eb-8daf-915ee5a82114.png)

I can't for the life of me figure out how to go about changing this, or if it's even possible.  Any help would be fantastic!

Thanks!

**Comment by greglandrum** [comment]
@bmcguir2 To follow your logic fully, the `+` should also be adjacent to the C, not with the H. This leads to renderings which are pretty ugly.

If you're trying to draw simple molecules for pedagogical purposes, you're probably better off not removing the Hs from the molecules and rendering the molecule with Hs present, like this:
```
>>> params = Chem.SmilesParserParams()
>>> params.removeHs = False
>>> m = Chem.MolFromSmiles('[C+][H]',params)
```
This yields the following drawing:
![image](https://user-images.githubusercontent.com/540511/125452595-b7dad5dc-7a8a-4c7a-843f-b831cc9fceac.png)

(one can argue about how long the bond is here relative to the font size, but that's a different problem)

**Reply by bmcguir2** [reply]
Hm; your code snippet on my local install still yields the original:

![image](https://user-images.githubusercontent.com/8889664/125620654-52ce30ab-98a9-47e3-b29a-f75fe72e60b6.png)

In any case, your point is sound that the logic would also dictate the + on the carbon, which is indeed sub-optimal.  I guess I'm not suggesting any change to default renderings, just whether there's an option or hack to move the electrons around to where I personally find them most illustrative?

**Reply by greglandrum** [reply]
apologies, I had copy-pasted the wrong line and forgot the version where I use the parameters. I've updated my previous comment.
